### DISCUSSION TITLE
Save journal data to iCloud

### DISCUSSION THREAD

**Discussion by olgv** [pajoma/vscode-journal#118] [General]
Hi there.

Is there any way one can point the `Journal:Base` towards an iCloud folder?
I made a Shortcut to a folder residing in iCloud, also tried adding the `~/Library/...` path to VS Code, but VS Code is unable to save the file in that location.

Any advice ?
Thank you.

**Comment by pajoma** [comment]
I am not sure how the iCloud folders are configured. I have tried it with Google Drive and OneDrive, both act like normal folders in the filesystem. 

Can you check the logs (switch on dev mode in journal settings, open "output" and select the journal in the dropdown

![grafik](https://user-images.githubusercontent.com/92156/134465766-441b52a4-9eb9-4295-8469-89b60800a4a1.png)

**Comment by dokie** [comment]
This works for me:

1. Create a soft link in your home folder to your iCloud folder where you want the files to live via (for example it could be called `MyJournal`) -  `ln -s ~/Library/Mobile\ Documents/com\~apple\~CloudDocs/MyJournal ~/Journal`
2. Make sure you use the default base dir then in the journal extension `~/Journal`

When I tried an alternative local folder it seemed not to work and saved journal files in the default local base dir.

**Comment by mameen** [comment]
I'm not sure about iCloud.. but I have multiple journals on multiple folders on OneDrive & GoogleDrive and they all work and sync.. I simple override the workspace JSON property `Journal:Base`

few things to look out for are:
* I check the option to [x] keep files locally
* Sync frequency is different and unpredictable! the best one that works in few seconds is OneDrive
* I have different work space for different platform-machine.. i.e. `myProject1-MyLaptop.workspace`, and ` `myProject1-MyPC.workspace`

**Comment by pajoma** [comment]
@olgv  Did the solutions by @dokie and @mameen work out for you?

**Comment by mameen** [comment]
There are many cloud file sync services that I already use. I even install git on top of it. I have no preference or suggestion other than not doing anything to keep the extension simple!
